applications/welcome/models/proc_harness.py

```diff
@@ -0,0 +1,22 @@
+#########################################################################
+## Define API for interacting with procedure table
+## These methods will be used in the controller, called by the code
+## that actually performs the sync to interact with the tables
+#########################################################################
+
+from gluon import current
+
+proc_table = current.db["procedures"]
+
```

Comment:
def do_synch():
    """Ask the server for all the procedures for this device_id, and get the latest version of each.  Write it in the proper files and update the tables."""